### PR TITLE
SPCTR-3032: Fix - Image - Setting up the hover image inside the container, chopping the image part

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

Image - Setting up the hover image inside the container, chopping the image part
Image will get chopped on the few sides

### Screenshots

https://d.pr/v/1YYrUt

### How has this been tested?

- Create a new page.
- Add Container of two columns.
- Add Heading on the left.
- Add Image on the right.
- Now setup the image “On Hover Image” as “Zoom In”.
- Check both backend and front-end.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +178,7 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 * Fix: Buttons - Resolved an issue in which the space between the icon and text would not apply in RTL sites.
 * Fix: Image - Resolved an issue in which the tablet and mobile border css would not be applied as intended.
 * Fix: Forms - Reworked the e-mail sending functionality to ensure that only one e-mail is sent when CC and BCC addresses are specified.
+* Fix: Image - Resolved an issue in which mask image is getting chopping off when zoomin is set.
```
Comment (bsf-zanev) [inline README.md:181]:
```suggestion
* Fix: Image - Resolved an issue in which mask images would get cut when the zoom-in hover effect was used.
```

**Review by bsf-zanev — APPROVED** [review]
Looks good, just a changelog update needed.

File: README.md (hunk 1 above)
Comment (rajmohanreddy774) [reply README.md]: Updated.

**Review by AbhijitNage123 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Review by aapatre — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -215,6 +215,9 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 
 ## Changelog ##
 
+### x.x.x ###
+* Fix: Image - Resolved an issue in which mask images would get cut when the zoom-in hover effect was used.
```
Comment (bsf-zanev) [inline README.md:219]:
```suggestion
* Fix: Image - Resolved an issue in which masked images would get cut-off when the zoom-in hover effect was used.
```

File: src/blocks/image/style.scss [hunk 3]
```diff
@@ -183,42 +184,48 @@
...
+			img,
+			.wp-block-uagb-image--layout-overlay__color-wrapper {
 				filter: blur(0);
 				transition: 0.35s ease-in-out;
 			}
 
 			&:hover {
-				img {
+				img,
+				.wp-block-uagb-image--layout-overlay__color-wrapper {
```
Comment (bsf-zanev) [inline src/blocks/image/style.scss:228]: Do we need to blur the overlay for this over effect?

**Review by bsf-zanev — APPROVED** [review]
Updates look good, just one query below.

File: includes/blocks/image/frontend.css.php [hunk 4]
```diff
@@ -223,7 +223,7 @@
 		$imagePath = $attr['maskCustomShape']['url'];
 	}
 	if ( ! empty( $imagePath ) ) {
-		$selectors['.wp-block-uagb-image .wp-block-uagb-image__figure img'] = array(
```
Comment (bsf-zanev) [inline includes/blocks/image/frontend.css.php:226]: Will this selector work as intended?

File: src/blocks/image/style.scss [hunk 5]
```diff
@@ -183,42 +184,48 @@
...
 						transform: translate3d(0, 0, 0);
 					}
 				}
 			}
 		}
 
 		&-grayscale {
-			img {
+			img,
+			.wp-block-uagb-image--layout-overlay__color-wrapper {
```
Comment (bsf-zanev) [inline src/blocks/image/style.scss:206]: I don't think we have to grayscale the overlay, just the image.

File: README.md (hunk 2 above)
Comment (rajmohanreddy774) [reply README.md]: Done.

File: src/blocks/image/style.scss (hunk 3 above)
Comment (rajmohanreddy774) [reply src/blocks/image/style.scss:228]: Yes as it is one of the options under hover effects.

File: includes/blocks/image/frontend.css.php (hunk 4 above)
Comment (rajmohanreddy774) [reply includes/blocks/image/frontend.css.php:226]: Yes it does.

File: src/blocks/image/style.scss (hunk 5 above)
Comment (rajmohanreddy774) [reply src/blocks/image/style.scss]: Removed and tested.

**Review by bsf-zanev — COMMENTED** [review]
Changes look good.

**Review by AbhijitNage123 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**